src/CMakeLists.txt

```diff
@@ -8,7 +8,7 @@ project(Rover LANGUAGES CXX C)
 find_package(OpenCV)
```

Comment:
How about something like this? This will set a reasonable default for `CMAKE_BUILD_TYPE` if the variable is not set explicitly. It also prints a message at the very beginning clearly stating what kind of build is being done.
```suggestion
# If a build type was not explicitly specified, default to "RelWithDebInfo" (Same optimization
# as the "Release" configuration but with debugging symbols).
# To change, you can pass -DCMAKE_BUILD_TYPE=... where ... is the configuration you would
# like to use. For example, if you want to do a debug build (little to no optimization and
# debugging symbols), add -DCMAKE_BUILD_TYPE=Debug to your `cmake` command.
if(NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING
      "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
endif()
message("========================================\n\
Building ${CMAKE_BUILD_TYPE} configuration...\n\
Add -DCMAKE_BUILD_TYPE=... to change\n\
========================================")

find_package(OpenCV)
```

Reply:
Otherwise everything looks good. I'm planning a CMake reorganization PR in the future so I could always include this with that instead.

Reply:
I think maybe save it for a future PR? This one's already kind of complicated.

Also, maybe it would be simpler just to put a note in the README at the cmake step letting people know about the `CMAKE_BUILD_TYPE` option? I don't mind much either way. But I think I read somewhere yesterday that the CMAKE_BUILD_TYPE setting is generally considered "the responsibility of the user"

Reply:
Ah okay, makes sense. I'm going to go ahead and merge this then